configure.sh

```diff
@@ -2,6 +2,120 @@
 
 # Variables
 BASE_DIR="$(realpath $(dirname ${BASH_SOURCE[0]}))"
+RHDH_GITHUB_INTEGRATION=${RHDH_GITHUB_INTEGRATION:-true}
+RHDH_GITLAB_INTEGRATION=${RHDH_GITLAB_INTEGRATION:-false}
+
+# Secret variables
+export GITHUB__APP__ID=${GITHUB__APP__ID:-''}
+export GITHUB__APP__CLIENT__ID=${GITHUB__APP__CLIENT__ID:-''}
+export GITHUB__APP__CLIENT__SECRET=${GITHUB__APP__CLIENT__SECRET:-''}
+export GITHUB__APP__WEBHOOK__URL=${GITHUB__APP__WEBHOOK__URL:-''}
+export GITHUB__APP__WEBHOOK__SECRET=${GITHUB__APP__WEBHOOK__SECRET:-''}
+export GITHUB__APP__PRIVATE_KEY=${GITHUB__APP__PRIVATE_KEY:-''}
+export GITOPS__GIT_TOKEN=${GITOPS__GIT_TOKEN:-''}
+export GITLAB__APP__CLIENT__ID=${GITLAB__APP__CLIENT__ID:-''}
+export GITLAB__APP__CLIENT__SECRET=${GITLAB__APP__CLIENT__SECRET:-''}
+export GITLAB__TOKEN=${GITLAB__TOKEN:-''}
+export QUAY__DOCKERCONFIGJSON=${QUAY__DOCKERCONFIGJSON:-''}
+export QUAY__API_TOKEN=${QUAY__API_TOKEN:-''}
+
+# Reads GitHub secrets if enabling GitHub integration
+if [[ $RHDH_GITHUB_INTEGRATION == "true" ]]; then
+    # Reads GitHub Org App ID
+    until [ ! -z "${GITHUB__APP__ID}" ]; do
+        read -p "Enter your GitHub App ID: " GITHUB__APP__ID
+        if [ -z "${GITHUB__APP__ID}" ]; then
+            echo "No GitHub App ID entered, try again."
+        fi
+    done
+
+    # Reads GitHub Org App Client ID
+    until [ ! -z "${GITHUB__APP__CLIENT__ID}" ]; do
+        read -p "Enter your GitHub App Client ID: " GITHUB__APP__CLIENT__ID
+        if [ -z "${GITHUB__APP__CLIENT__ID}" ]; then
+            echo "No GitHub App Client ID entered, try again."
+        fi
+    done
+
+    # Reads GitHub Org App Client Secret
+    until [ ! -z "${GITHUB__APP__CLIENT__SECRET}" ]; do
+        read -p "Enter your GitHub App Client Secret: " GITHUB__APP__CLIENT__SECRET
+        if [ -z "${GITHUB__APP__CLIENT__SECRET}" ]; then
+            echo "No GitHub App Client Secret entered, try again."
+        fi
+    done
+
+    # Reads GitHub Org App Webhook URL
+    until [ ! -z "${GITHUB__APP__WEBHOOK__URL}" ]; do
```

Comment:
If I don't run `bash ./scripts/configure-pipelines.sh` before then I don't have a webhook URL available to me to input here

Reply:
Yeah I noticed that too. I just input it manually because I knew what it would be!

Reply:
Why does the user need to input the webhook url env?

Reply:
We could probably structure the script so that the webhook url is grabbed from the secret through the script (since the pipelines config script creates this secret i believe) and then users don't need to go find it? wdyt @michael-valdron

Reply:
>> Why does the user need to input the webhook url env?
> We could probably structure the script so that the webhook url is grabbed from the secret through the script (since the pipelines config script creates this secret i believe) and then users don't need to go find it? wdyt @michael-valdron

The script could fetch the Webhook URL from the route like I did with the Backstage `baseUrl` patching.

```sh
WEBHOOK_URL="https://$(kubectl get route -n openshift-pipelines pipelines-as-code-controller --ignore-not-found -o jsonpath={.spec.host})"
```

@maysunfaisal @Jdubrick WDYT?

Reply:
It could even print to the console to show the user what to put into the GitHub App field.

Reply:
> since the pipelines config script creates this secret i believe

No, the pipelines config script creates a secret that stores the `GITHUB__APP__WEBHOOK__SECRET` not the `GITHUB__APP__WEBHOOK__URL`.

Reply:
+1 I like the idea of fetching it within the `configure.sh` if you decide to use that instead of running the 3 setup scripts manually, and printing it out will also be helpful!

Does it not create the secret containing the url here? https://github.com/redhat-ai-dev/ai-rhdh-installer/blob/main/scripts/configure-pipelines.sh#L139-#L143

@michael-valdron

Reply:
I think we need to print this info https://github.com/redhat-ai-dev/ai-rhdh-installer/pull/10#issuecomment-2402874415 because users will need to also update their GitHub application. A total new person will not know how to do this unless installer guides them

Reply:
> Does it not create the secret containing the url here? https://github.com/redhat-ai-dev/ai-rhdh-installer/blob/main/scripts/configure-pipelines.sh#L139-#L143
> 
> @michael-valdron

@Jdubrick Ah you right it is already being fetched on [line 143](https://github.com/redhat-ai-dev/ai-rhdh-installer/blob/753af9a15690059aa3f1f2eba5ce6d5f7efa671e/scripts/configure-pipelines.sh#L143) and set into the secret, didn't notice that my bad.

> +1 I like the idea of fetching it within the `configure.sh` if you decide to use that instead of running the 3 setup scripts manually, and printing it out will also be helpful!

With the above, I'm thinking the quickest way to do this is I can add this step in both pipelines and rhdh config scripts with control logic to determine if the secret has been created yet, if so just fetch the webhook url from the secret and if not create the secret as it currently does.

Reply:
> I think we need to print this info [#10 (comment)](https://github.com/redhat-ai-dev/ai-rhdh-installer/pull/10#issuecomment-2402874415) because users will need to also update their GitHub application. A total new person will not know how to do this unless installer guides them

Yeah I agree that anything we need from the cluster should be printed out so the user can use for the GitHub app fields.

Reply:
+1 to both points! I like those ideas

Reply:
> > I think we need to print this info [#10 (comment)](https://github.com/redhat-ai-dev/ai-rhdh-installer/pull/10#issuecomment-2402874415) because users will need to also update their GitHub application. A total new person will not know how to do this unless installer guides them
> 
> Yeah I agree that anything we need from the cluster should be printed out so the user can use for the GitHub app fields.

Done in https://github.com/redhat-ai-dev/ai-rhdh-installer/pull/10/commits/3a820df5e477b1d8c80c23fb8bdd2951e9527afe

FYI @maysunfaisal

Reply:
> > Does it not create the secret containing the url here? https://github.com/redhat-ai-dev/ai-rhdh-installer/blob/main/scripts/configure-pipelines.sh#L139-#L143
> > @michael-valdron
> 
> @Jdubrick Ah you right it is already being fetched on [line 143](https://github.com/redhat-ai-dev/ai-rhdh-installer/blob/753af9a15690059aa3f1f2eba5ce6d5f7efa671e/scripts/configure-pipelines.sh#L143) and set into the secret, didn't notice that my bad.
> 
> > +1 I like the idea of fetching it within the `configure.sh` if you decide to use that instead of running the 3 setup scripts manually, and printing it out will also be helpful!
> 
> With the above, I'm thinking the quickest way to do this is I can add this step in both pipelines and rhdh config scripts with control logic to determine if the secret has been created yet, if so just fetch the webhook url from the secret and if not create the secret as it currently does.

Provided in https://github.com/redhat-ai-dev/ai-rhdh-installer/pull/10/commits/d14cd482e6b67a1f54fe53fd3c76e78a5a81c4df, with a fixed control logic in https://github.com/redhat-ai-dev/ai-rhdh-installer/pull/10/commits/4ec8f7a3b170dae6eb33f22e7c49abff4bb48e7b

FYI @maysunfaisal @Jdubrick